SimulationAgent/DeviceConnection/Disconnect.cs

```diff
@@ -50,22 +52,22 @@ public async Task RunAsync()
 
             try
             {
-                this.deviceConnectionActor.Client = this.deviceConnectionActor.SimulationContext.Devices.GetClient(
-                    this.deviceConnectionActor.Device,
+                this.deviceContext.Client = this.simulationContext.Devices.GetClient(
```

Comment:
this is strange, probably wrong. When disconnecting a device there should already be a client in the device context, which is the client to be disconnected. This looks like we're getting a new client, which will be in the default state (i.e. disconnected) and ask it to disconnect

Reply:
Good catch. This is out of scope for this change, so I've added a TODO.